### PR TITLE
Fix semicolon must decrease special indentation level

### PR DESCRIPTION
**Pull request by mvorisek**
Some DBs allows multiple statements per single query and the formatter must support such syntax.

### CODE REVIEW

**Comment by mvorisek** [comment]
@greg0ire can I have your review here? Thank you!

**Review by greg0ire — APPROVED** [review]

**Comment by greg0ire** [comment]
Please squash your commits.

**Comment by mvorisek** [comment]
Squashed.

Once merged, please merge the 1.4.x into 1.5.x like in https://github.com/doctrine/sql-formatter/pull/111/commits/4a151ddca07c7d9c71f6b134b9fa7e54749b1e7c.

**Review by SenseException — APPROVED** [review]

File: src/SqlFormatter.php [hunk 1]
```diff
@@ -256,6 +256,15 @@ public function format(string $string, string $indentString = '  '): string
                 if ($token->value() === 'LIMIT' && ! $inlineParentheses) {
                     $clauseLimit = true;
                 }
+            } elseif ($token->value() === ';') {
```
Comment (goetas) [inline src/SqlFormatter.php:259]: the inner code of this if branch is almost identical to the `} elseif ($token->isOfType(Token::TOKEN_TYPE_RESERVED_TOPLEVEL)) {` few lines before. 
would it make sense to group them?

**Review by goetas — CHANGES_REQUESTED** [review]

File: src/SqlFormatter.php (hunk 1 above)
Comment (mvorisek) [reply src/SqlFormatter.php:259]: It is not worth grouping them in one if, but see #111, I have already started addressing such code duplication there. This PR should stay as is as no easy deduplication is possible and it is a bugfix.

Comment (mvorisek) [reply src/SqlFormatter.php:259]: @goetas can you please approve this PR?

**Comment by mvorisek** [comment]
@greg0ire can this PR be please merged + merged up so I can work on the other PRs? Thank you.

**Comment by greg0ire** [comment]
@mvorisek for some reason there are loads of conflicts, can you please send a PR to address them?

**Comment by mvorisek** [comment]
@greg0ire sure, here it is #114, please merge, I will then rebase #111

File: src/SqlFormatter.php (hunk 1 above)
Comment (goetas) [reply src/SqlFormatter.php:259]: yea, i'm sorry, I did not see how long/complex was that if branch

**Pull request merged**